Framework/Algorithms/src/MonitorEfficiencyCorUser.cpp

```diff
@@ -97,8 +100,8 @@ void MonitorEfficiencyCorUser::exec() {
  * @param eOut corrected spectrum errors
  */
 void MonitorEfficiencyCorUser::applyMonEfficiency(
-    const size_t numberOfChannels, const MantidVec &yIn, const MantidVec &eIn,
-    const double effVec, MantidVec &yOut, MantidVec &eOut) {
+    const size_t numberOfChannels, const HistogramY &yIn, const HistogramE &eIn,
+    const double effVec, HistogramY &yOut, HistogramE &eOut) {
 
   for (unsigned int j = 0; j < numberOfChannels; ++j) {
     yOut[j] = yIn[j] / effVec;
```

Comment:
This could be made much more efficient by computing `1 / effVec` in advance (why is it called `Vec`??). And actually this is just `operator*=` (you can also implement `operator/=` could be useful.

And we can just go one step further, and finally implement arithmetics for `Histogram`, then we can completely remove this function and the whole `yOut`, `yIn`, ... bla bla above:

```cpp
Histogram operator*(Histogram lhs, const double rhs) {
  lhs.mutableY() *= rhs;
  lhs.mutableE() *= rhs;
  return lhs;
}
```

Then the whole bunch of code in `MonitorEfficiencyCorUser` can be replaced by:

```cpp
m_outputWS->setHistogram(i, m_inputWS->histogram(i) * (1.0/eff0));
```

I think.